internal/indexer/indexer.go

```diff
@@ -480,3 +481,18 @@ func WarehousesBySubscribedURLs(obj client.Object) []string {
 	}
 	return repoURLs
 }
+
+// ProjectConfigsByWebhookReceiverPaths is a client.IndexerFunc that indexes Projects by the
+// paths of their receivers.
+func ProjectConfigsByWebhookReceiverPaths(obj client.Object) []string {
+	pc, ok := obj.(*kargoapi.ProjectConfig)
+	if !ok {
+		return nil
+	}
+
+	var receiverPaths []string
+	for _, r := range pc.Status.WebhookReceivers {
+		receiverPaths = append(receiverPaths, r.Path)
+	}
```

Comment:
We know the size of the final list up front so we can avoid lots of expensive resizing this way:

```suggestion
	receiverPaths := make([]string, len(pc.Status.WebhookReceivers))
	for i, r := range pc.Status.WebhookReceivers {
		receiverPaths[i] = r.Path
	}
```